### ISSUE TITLE
DROID CLI can't identify file format by extension in no-profile-mode

### ISSUE THREAD

**Issue by toKrause** [digital-preservation/droid#224]
The current signature file (`DROID_SignatureFile_V94.xml`) contains some file format definitions that contain only an `<Extension>` and no `<InternalSignatureID>`-reference to a byte signature. One such example is the file format definition  for "Java language source code file" (`x-fmt/422`):

```
<FileFormat ID="809" Name="Java language source code file" PUID="x-fmt/422">
   <Extension>java</Extension>
</FileFormat>
```
DROID UI identifies a file with the `.java`-extension correctly as `x-fmt/422` but DROID CLI returns `Unknown`.

---

Input:
```
touch Test.java
./droid.sh -Nr Test.java -Ns DROID_SignatureFile_V94.xml -Nc container-signature-20180920.xml
```

Output:
```
DROID 6.4 No Profile mode: Runtime Information
Selected folder or file: Test.java
Binary signature file: DROID_SignatureFile_V94.xml
Container signature file: container-signature-20180920.xml
Open archives: False
Open web archives: False
/home/xxx/DROID-6.4/Test.java,Unknown
```

**Comment by alexgreenDP** [comment]
Hello Torsten,

Thank you for raising this - it's really useful to know. We'll investigate this functionality for potential inclusion in a future version of DROID.

Alex

**Comment by toKrause** [comment]
> We'll investigate this functionality for potential inclusion in a future version of DROID.

Well, I haven't really suggested a new feature to be added.

**Comment by marhop** [comment]
Hi,

Will this be fixed anytime soon? It's a little annoying that DROID CLI can't even identify a plain text file like foo.txt (result Unknown) when DROID GUI returns a correct PUID (result x-fmt/111).

Thanks,
Martin

**Comment by kalmicka** [comment]
Has this been fixed by any chance? Using droid6.4 and still getting some "unknown" responses.

**Comment by jcharlet** [comment]
a bit late to come to the party but going to look at this, see if it can be included in the upcoming release.

**Comment by jcharlet** [comment]
ok we analyzed it with @sparkhi , the file extension identification is implemented in the GUI, but missing in the CLI, which explains the inconsistency of the results between gui and cli. 

we could either reimplement the extension identification in the CLI, or refactor the whole identification mechanism as @nishihatapalmer is already looking at. Either way, there is no quick win, it's a significant feature so we won't provide this for 6.5 release.

**Comment by nishihatapalmer** [comment]
This is fixed by some work I've done to centralise identification across DROID.  Won't be merged into the next release I think, but once we have this, DROID won't have differences in identification between command line and the GUI.

**Comment by jcharlet** [comment]
to be fixed with https://github.com/digital-preservation/droid/issues/470

**Comment by sparkhi** [comment]
Fixed in 6.6.0
